integration/manualtests/tx_test.go

```diff
@@ -67,3 +77,60 @@ func TestIssueTxWaitReceipt(t *testing.T) {
 		t.Fatalf("Did not mine the transaction after %s seconds  - receipt: %+v", 30*time.Second, receipt)
 	}
 }
+
+func TestL2IssueTxWaitReceipt(t *testing.T) {
+	// test is skipped by default to avoid breaking CI - set env flag in `Run Configurations` to run it in IDE
+	if os.Getenv(_IDEFlag) == "" {
+		t.Skipf("set flag %s to run this test in the IDE", _IDEFlag)
+	}
+
+	ctx := context.Background()
+	w := wallet.NewInMemoryWalletFromConfig(
+		"4bfe14725e685901c062ccd4e220c61cf9c189897b6c78bd18d7f51291b2b8f8",
+		777,
+		gethlog.New())
+	host := "dev-testnet.obscu.ro"
+	port := 13001
+
+	vk, err := rpc.GenerateAndSignViewingKey(w)
+	assert.Nil(t, err)
+	client, err := rpc.NewEncNetworkClient(fmt.Sprintf("ws://%s:%d", host, port), vk, gethlog.New())
+	assert.Nil(t, err)
+	authClient := obsclient.NewAuthObsClient(client)
+
+	toAddr := datagenerator.RandomAddress()
+	nonce, err := authClient.NonceAt(ctx, nil)
+	assert.Nil(t, err)
+
+	w.SetNonce(nonce)
+	estimatedTx := authClient.EstimateGasAndGasPrice(&types.LegacyTx{
+		Nonce: w.GetNonceAndIncrement(),
+		To:    &toAddr,
+		Value: big.NewInt(100),
+	})
+	assert.Nil(t, err)
+
+	signedTx, err := w.SignTransaction(estimatedTx)
+	assert.Nil(t, err)
+
+	err = authClient.SendTransaction(ctx, signedTx)
+	assert.Nil(t, err)
+
+	fmt.Printf("Created Tx: %s \n", signedTx.Hash().Hex())
+	fmt.Printf("Checking for tx receipt for %s \n", signedTx.Hash())
+	var receipt *types.Receipt
+	for start := time.Now(); time.Since(start) < time.Minute; time.Sleep(time.Second) {
+		receipt, err = authClient.TransactionReceipt(ctx, signedTx.Hash())
+		if err == nil {
+			break
+		}
+		//if !errors.Is(err, ethereum.NotFound) {
```

Comment:
should this be uncommented/deleted?

Reply:
That's a bug I found actually - but you're right it needs more info